Library/Statements/TryCatchStatement.php

```diff
@@ -97,28 +101,34 @@ public function compile(CompilationContext $compilationContext)
                  * Check if any of the classes in the catch block match the thrown exception
                  */
                 foreach ($catch['classes'] as $class) {
-                    $ifCheck = $exprBuilder->statements()->ifX()
+                    $ifs[] = $exprBuilder->statements()->ifX()
                         ->setCondition(
                             $exprBuilder->operators()->binary(
                                 BinaryOperator::OPERATOR_INSTANCEOF,
-                                $exprBuilder->variable($variable->getName()),
+                                $exprBuilder->variable($exc_var->getName()),
                                 $exprBuilder->variable($class['value'])
                             )
                         )
                         ->setStatements($exprBuilder->statements()->block(array_merge(
-                            array($exprBuilder->statements()->rawC('zend_clear_exception(TSRMLS_C);')),
+                            array(
+                                $exprBuilder->statements()->rawC('zend_clear_exception(TSRMLS_C);'),
+                                $exprBuilder->statements()->rawC('ZEPHIR_CPY_WRT(&' . $variable->getName() . ', &' . $exc_var->getName() . ');'),
```

Comment:
Is there a cleaner way?

`$compilationContext->backend->copyOnWrite()` writes the code, what I need is the assignment operator as this probably won't compile on ZE2